[AOTInductor] Add model runner to avoid using torch_extension

khabinov opened the pull request:
Differential Revision: D49609669




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

chenyang78 reviewed (approved):
Please fix the lint issues. Otherwise, LGTM. Thanks!

khabinov committed:
[AOTInductor] Add model runner to avoid using torch_extension (#110263)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/110263

Reviewed By: chenyang78, SherlockNoMad

Differential Revision: D49609669
